static/app/utils/panelProvider.tsx

```diff
@@ -0,0 +1,19 @@
+import {ThemeProvider, useTheme} from '@emotion/react';
+
+const smallerBorderRadii = {
+  borderRadius: '4px',
+  borderRadiusBottom: '0 0 4px 4px',
+  borderRadiusTop: '4px 4px 0 0',
+  borderRadiusLeft: '4px 0 0 4px',
+  borderRadiusRight: '0 4px 4px 0',
+};
+
+const PanelProvider = ({children}: {children: React.ReactNode}) => {
+  const theme = useTheme();
+
+  return (
+    <ThemeProvider theme={{...theme, ...smallerBorderRadii}}>{children}</ThemeProvider>
+  );
+};
+
+export default PanelProvider;
```

Comment:
This nested theme provider enforces the new rule, making sure that descendants of containers (`Overlay`, `Panel`) will have smaller border radii. This way, we don't need to worry about what value to use. We can just get `theme.borderRadius` and it'll just work.

Reply:
https://emotion.sh/docs/theming#themeprovider-reactcomponenttype

Reply:
I love this solution, very elegant. Can you add a detailed comment here that outlines the reasoning behind this? Since it works like magic we should leave some breadcrumbs for those who might be curious in the future.